lekko_client/stores/store.py

```diff
@@ -0,0 +1,60 @@
+from abc import ABC, abstractmethod
+from hashlib import sha256
+
+from lekko_client.gen.lekko.backend.v1beta1.distribution_service_pb2 import (
+    GetRepositoryContentsResponse,
+)
+from lekko_client.models import FeatureData
+
+
+class Store(ABC):
+    def __init__(self):
+        self._commit_sha = ""
+        self._content_hash = ""
+
+    @abstractmethod
+    def get(self, namespace: str, config_key: str) -> FeatureData:
+        ...
+
+    def load(self, contents: GetRepositoryContentsResponse) -> bool:
+        if not contents:
+            return False
+
+        contents = self.sort_contents(contents)
+        content_hash = self.hash_contents(contents)
+
+        if not self.should_update(contents, content_hash):
```

Comment:
Don't we need to have a lock here? Event if updating contents is atomic, commit sha and content_hash could be our of sync.

Reply:
yeah, that's on the todo list. i should document what's left to do in this pr.

Reply:
oh, sorry maybe I jumped to review it too early :( ? then just let me know when it's ready for the next round of review

Reply:
sorry - should have been clear. we're not 100% feature complete, but this is ready for feedback on the interface/structure.  We'll be adding more robustness here, plus tests, and the event logging.